src/main/java/org/example/ClientHandler.java

```diff
@@ -0,0 +1,42 @@
+package org.example;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.Socket;
+
+public class ClientHandler extends Thread {
```

Comment:
If this class is responsible for actual echo logic, lets call it something like `EchoService` or `EchoMessageBus`

Reply:
Changed `ClientHandler` to `EchoMessageBus` in latest commit.